module/applications/dialogs/beastformDialog.mjs

```diff
@@ -278,19 +278,30 @@ export default class BeastformDialog extends HandlebarsApplicationMixin(Applicat
                 'close',
                 async () => {
                     const selected = app.selected.toObject();
+                    const evolved = app.evolved.form ? app.evolved.form.toObject() : null;
                     const data = await game.system.api.data.items.DHBeastform.getWildcardImage(
                         app.configData.data.parent,
-                        app.selected
+                        evolved ?? app.selected
                     );
                     if (data) {
                         if (!data.selectedImage) selected = null;
                         else {
-                            if (data.usesDynamicToken) selected.system.tokenRingImg = data.selectedImage;
-                            else selected.system.tokenImg = data.selectedImage;
+                            if (evolved) {
+                                if (evolved.usesDynamicToken) evolved.system.tokenRingImg = data.selectedImage;
+                                else evolved.system.tokenImg = data.selectedImage;
+                            } else {
+                                if (data.usesDynamicToken) selected.system.tokenRingImg = data.selectedImage;
+                                else selected.system.tokenImg = data.selectedImage;
+                            }
```

Comment:
these two branches look about the same, other than the first branch checks evolved and assigns evolved, and the second checks data and assigns selected. There a way to consolidate?